src/reachy2_sdk/parts/mobile_base.py

```diff
@@ -0,0 +1,284 @@
+from reachy2_sdk_api.mobile_base_mobility_pb2 import (
+    DirectionVector,
+    GoToVector,
+    TargetDirectionCommand,
+)
+from reachy2_sdk_api.mobile_base_mobility_pb2_grpc import MobileBaseMobilityServiceStub
+from reachy2_sdk_api.mobile_base_utility_pb2 import (
+    ControlModeCommand,
+    ControlModePossiblities,
+)
+from reachy2_sdk_api.mobile_base_utility_pb2 import MobileBase as MobileBase_proto
+from reachy2_sdk_api.mobile_base_utility_pb2 import (
+    MobileBaseState,
+    ZuuuModeCommand,
+    ZuuuModePossiblities,
+)
+from reachy2_sdk_api.mobile_base_utility_pb2_grpc import MobileBaseUtilityServiceStub
+
+from ..subparts.lidar import Lidar
+
+
+class MobileBase:
+    """The MobileBaseSDK class handles the connection with Reachy's mobile base.
+
+    It holds:
+
+    - the odometry of the base (you can also easily reset it),
+    - the battery voltage to monitor the battery usage,
+    - the control and drive mode of the base,
+    - two methods to send target positions or target velocities.
+
+    If you encounter a problem when using the base, you have access to an emergency shutdown method.
+    """
+
+    def __init__(
+        self,
+        mb_msg: MobileBase_proto,
+        initial_state: MobileBaseState,
+        grpc_channel: grpc.Channel,
+    ) -> None:
+        """Set up the connection with the mobile base."""
+        self._utility_stub = MobileBaseUtilityServiceStub(grpc_channel)
+        self._mobility_stub = MobileBaseMobilityServiceStub(grpc_channel)
+
+        self._drive_mode = self._get_drive_mode().lower()
+        self._control_mode = self._get_control_mode().lower()
+
+        self._max_xy_vel = 1.0
+        self._max_rot_vel = 180.0
+        self._max_xy_goto = 1.0
+
+        self.lidar = Lidar(initial_state.lidar_obstacle_detection_status, grpc_channel)
+
+        self._update_with(initial_state)
+
+    def __repr__(self) -> str:
+        """Clean representation of a mobile base."""
+        repr_template = (
+            '<MobileBase host="{host}" on={on} \n'
+            " lidar_safety_enabled={lidar_safety_enabled} \n"
+            " battery_voltage={battery_voltage}>"
+        )
+        return repr_template.format(
+            on=self.is_on(),
+            lidar_safety_enabled=self.lidar.safety_enabled,
+            battery_voltage=self.battery_voltage,
+        )
+
+    def _get_drive_mode(self) -> ZuuuModeCommand:
+        mode_id = self._utility_stub.GetZuuuMode(Empty()).mode
+        return ZuuuModePossiblities.keys()[mode_id]
+
+    def _get_control_mode(self) -> ControlModePossiblities:
+        mode_id = self._utility_stub.GetControlMode(Empty()).mode
+        return ControlModePossiblities.keys()[mode_id]
+
+    @property
+    def battery_voltage(self) -> float:
+        """Return the battery voltage. Battery should be recharged if it reaches 24.5V or below."""
+        return float(round(self._utility_stub.GetBatteryLevel(Empty()).level.value, 1))
```

Comment:
We can print a warning if battery_voltage is below 24.5. What do you think?

Reply:
Added in #294